exercises/practice/spiral-matrix/.meta/example.c

```diff
@@ -0,0 +1,50 @@
+#include <stdlib.h>
+#include "spiral_matrix.h"
+
+spiral_matrix_t *spiral_matrix_create(int size)
+{
+   spiral_matrix_t *spiral_matrix = calloc(1, sizeof(spiral_matrix_t));
+   spiral_matrix->size = size;
+
+   if (size == 0) {
+      return spiral_matrix;
+   }
+
+   spiral_matrix->matrix = calloc(size, sizeof(int *));
+   for (int i = 0; i < size; i++) {
+      spiral_matrix->matrix[i] = calloc(size, sizeof(int));
+   }
+
+   int x = 0;
+   int y = 0;
+   int dx = 1;
+   int dy = 0;
+
+   for (int i = 1; i <= size * size; i++) {
```

Comment:
`size_t i` maybe?

Reply:
I started with `size_t`, actually. The reason I changed to `int` is that I want to compare to `x + dx` and `y + dy` to `size` with a minimum of awkwardness and with `size_t` there's a signedness conflict. Casting `x + dx` to `size_t` isn't safe because `size_t` can't represent when `x + dx` is negative, casting `size` up to a larger signed type isn't an option, and casting `size` down to an `int` isn't much different than using `int` in the first place. I would use `ssize_t`, but this isn't included in standard C. C++ has standardized `std::ssize` and widespread use of unsigned size types is considered to have been a mistake in the C++ standard library.

In other words, I agree with you in principle, but this felt like the best of the bad options and least likely to cause awkward situations for students.